eslint.config.mjs

```diff
@@ -306,6 +306,16 @@ export default [
           selector: "NewExpression[callee.name='Ref']",
           message: "Use `Ref.get()` rather than `new Ref()`.",
         },
+        {
+          selector: "ExportNamedDeclaration[declaration]",
+          message:
+            "Separate the declaration and the export statement, using `export { ... }`.",
+        },
+        {
+          selector: "ExportDefaultDeclaration:has(> :declaration)",
+          message:
+            "Separate the declaration and the export statement, using `export default <variable name>`.",
+        },
```

Comment:
Possibly I'm not understanding the selector here, but given the message I'd expect that e.g. the following case would need an `eslint-disable` statement?
https://github.com/mozilla/pdf.js/blob/72feb4c25694d1e4dcfe44a3243a5c91605be513/eslint.config.mjs#L26

Reply:
`:declaration` covers functions/classes/variables (and import/export, which can never appear inside an `export default`).

The error message is because if you have an invalid `export default function x() {}` you are going to split it to `function x() {}`+`export default x`, using the variable name in `export default`.